### ISSUE TITLE
Model not saved

### ISSUE THREAD

**Issue by shaihr** [taolei87/rcnn#9]
Hi,
I tried to train a model using rationale.py on a small dataset (with not rationale 'gold data'). It seems that the model is able to learn, and the loss decreases from one epoch to another. However, when the training ended the model wasn't saved and I wasn't able to use it.
How can I save the model? and how can it be loaded and used to classify new documents?

The command I use to train the model is:

python rationale\rationale.py --embedding <word2vec path> --train <training file> --dev <dev file> --save_model model.pkl.gz --batch 50

Thanks,
Shai

**Comment by taolei87** [comment]
Hi @shaihr  
model saving for later usage is not implemented. I'm traveling during thanksgiving holidays. Let me get this back to you next week.

**Comment by taolei87** [comment]
Hi @shaihr 

I added code to save and load model in [rationale/rationale.py](../blob/master/code/rationale/rationale.py) and [ratioanle/rationale_dependent.py](../blob/master/code/rationale/rationale_dependent.py). See this [diff](https://github.com/taolei87/rcnn/commit/5f78b0d57455bd15be0e5158a019b51958f522eb). 
In particular, [this part](https://github.com/taolei87/rcnn/blob/master/code/rationale/rationale.py#L620-L658) and [this part](https://github.com/taolei87/rcnn/blob/master/code/rationale/rationale_dependent.py#L657-L670) are examples of using the saved model.

**Note:** 
You still need to provide pre-trained word embeddings. The embeddings are fixed and are not saved to the model file.

**Comment by shaihr** [comment]
Thanks for the quick solution!
